crates/biome_js_analyze/src/lint/nursery/use_explicit_function_return_type.rs

```diff
@@ -0,0 +1,158 @@
+use biome_rowan::{declare_node_union, AstNode, TextRange};
+
+declare_lint_rule! {
+    /// Require explicit return types on functions and class methods.
+    ///
+    /// Functions in TypeScript often don't need to be given an explicit return type annotation.
+    /// Leaving off the return type is less code to read or write and allows the compiler to infer it from the contents of the function.
+    ///
+    /// However, explicit return types do make it visually more clear what type is returned by a function.
+    /// They can also speed up TypeScript type checking performance in large codebases with many large functions.
+    /// Explicit return types also reduce the chance of bugs by asserting the return type, and it avoids surprising "action at a distance," where changing the body of one function may cause failures inside another function.
+    ///
+    /// This rule enforces that functions do have an explicit return type annotation.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```ts,expect_diagnostic
+    /// // Should indicate that no value is returned (void)
+    /// function test() {
+    ///   return;
+    /// }
+    /// ```
+    ///
+    /// ```ts,expect_diagnostic
+    /// // Should indicate that a number is returned
+    /// var fn = function () {
+    ///    return 1;
+    /// };
+    /// ```
+    ///
+    /// ```ts,expect_diagnostic
+    /// // Should indicate that a string is returned
+    /// var arrowFn = () => 'test';
+    /// ```
+    ///
+    /// ```ts,expect_diagnostic
+    /// class Test {
+    ///   // Should indicate that no value is returned (void)
+    ///   method() {
+    ///     return;
+    ///   }
+    /// }
+    /// ```
+    ///
+    /// ```ts,expect_diagnostic
+    /// // Should indicate that no value is returned (void)
+    /// function test(a: number) {
+    ///   a += 1;
+    /// }
+    /// ```
+    ///
+    /// ### Valid
+    /// ```ts
+    /// // No return value should be expected (void)
+    /// function test(): void {
+    ///   return;
+    /// }
+    /// ```
+    ///
+    /// ```ts
+    /// // A return value of type number
+    /// var fn = function (): number {
+    ///   return 1;
+    /// }
+    /// ```
+    ///
+    /// ```ts
+    /// // A return value of type string
+    /// var arrowFn = (): string => 'test';
+    /// ```
+    ///
+    /// ```ts
+    /// class Test {
+    ///   // No return value should be expected (void)
+    ///   method(): void {
+    ///     return;
+    ///   }
+    /// }
+    /// ```
+    ///
+    pub UseExplicitFunctionReturnType {
+        version: "next",
+        name: "useExplicitFunctionReturnType",
+        language: "ts",
+        recommended: false,
+        sources: &[RuleSource::EslintTypeScript("explicit-function-return-type")],
+    }
+}
+
+declare_node_union! {
+    pub AnyJsFunctionAndMethod = AnyJsFunction | JsMethodClassMember | JsGetterClassMember
```

Comment:
We could add `JsMethodObjectMember` and `JsGetterObjectMember`?

Also, I suggest renaming the type to `AnyJsFunctionWithReturnType` because in the code base we already have types named `AnyFunctionLike` and `AnyJsFunctionOrMethod`. I am a bit afraid of confusion.

```suggestion
    pub AnyJsFunctionWithReturnType = AnyJsFunction | JsMethodClassMember | JsMethodObjectMember | JsGetterClassMember | JsGetterObjectMember
```

Reply:
I’ve added support for JsMethodObjectMember and JsGetterObjectMember, and I’ve renamed the type to AnyJsFunctionWithReturnType as suggested. I’ve also added test cases for object members to ensure everything is covered!!